packages/trip-viewer-overlay/src/index.js

```diff
@@ -0,0 +1,46 @@
+import { encodedPolylineType } from "@opentripplanner/core-utils/lib/types";
+import PropTypes from "prop-types";
+import React from "react";
+import { FeatureGroup, MapLayer, Polyline, withLeaflet } from "react-leaflet";
+
+import polyline from "@mapbox/polyline";
+
+class TripViewerOverlay extends MapLayer {
+  componentDidMount() {}
+
+  // TODO: determine why the default MapLayer componentWillUnmount() method throws an error
+  componentWillUnmount() {}
+
+  componentDidUpdate(prevProps) {
+    const oldGeometry = prevProps.tripData && prevProps.tripData.geometry;
+    const newGeometry = this.props.tripData && this.props.tripData.geometry;
+    if (oldGeometry === newGeometry || !newGeometry) return;
+    const pts = polyline.decode(newGeometry.points);
+    this.props.leaflet.map.fitBounds(pts);
+  }
+
+  createLeafletElement() {}
+
+  updateLeafletElement() {}
+
+  render() {
+    const { tripData } = this.props;
+
+    if (!tripData || !tripData.geometry) return <FeatureGroup />;
+
+    const pts = polyline.decode(tripData.geometry.points);
+    return (
+      <FeatureGroup>
+        <Polyline positions={pts} weight={8} color="#00bfff" opacity={0.6} />
```

Comment:
Should the weight, color, and opacity be props? Maybe we should pass a `polylineProps` that contains the default values shown above that can be overridden?